From 80a01a39dbd7a2222e30adf00f6d5019b1f4bba7 Mon Sep 17 00:00:00 2001
From: Saikrishna Arcot <sarcot@microsoft.com>
Date: Thu, 13 Oct 2022 13:45:17 -0700
Subject: [PATCH 2/3] Revert commit 69334996: make
 sshd_config:ClientAliveCountMax=0 disable the connection-killing behavior

SONiC (and others) use this feature to kill connections when the session
is idle after some duration of time. OpenSSH 8.2 defined setting
ClientAliveCountMax=0, but by doing so, broke the current use case of
it.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
---
 serverloop.c  | 3 +--
 sshd_config.5 | 3 ---
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/serverloop.c b/serverloop.c
index 6db0916d8..6ab7e2312 100644
--- a/serverloop.c
+++ b/serverloop.c
@@ -134,8 +134,7 @@ client_alive_check(struct ssh *ssh)
 	int r, channel_id;
 
 	/* timeout, check to see how many we have had */
-	if (options.client_alive_count_max > 0 &&
-	    ssh_packet_inc_alive_timeouts(ssh) >
+	if (ssh_packet_inc_alive_timeouts(ssh) >
 	    options.client_alive_count_max) {
 		sshpkt_fmt_connection_id(ssh, remote_id, sizeof(remote_id));
 		logit("Timeout, client not responding from %s", remote_id);
diff --git a/sshd_config.5 b/sshd_config.5
index 8e0b58ebf..9d1392f24 100644
--- a/sshd_config.5
+++ b/sshd_config.5
@@ -621,9 +621,6 @@ is set to 15, and
 .Cm ClientAliveCountMax
 is left at the default, unresponsive SSH clients
 will be disconnected after approximately 45 seconds.
-Setting a zero
-.Cm ClientAliveCountMax
-disables connection termination.
 .It Cm ClientAliveInterval
 Sets a timeout interval in seconds after which if no data has been received
 from the client,
-- 
2.43.0

